compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -2213,7 +2214,10 @@ fn determine_place_ancestry_relation<'tcx>(
 ///
 /// Reason we only drop the last deref is because of the following edge case:
 ///
-/// ```rust
+/// ```ignore HELP (I feel like this can be made to work but I'm doing something wrong)
```

Comment:
try this:
```diff
diff --git a/compiler/rustc_typeck/src/check/upvar.rs b/compiler/rustc_typeck/src/check/upvar.rs
index aef6c2b68ab..d51588d2737 100644
--- a/compiler/rustc_typeck/src/check/upvar.rs
+++ b/compiler/rustc_typeck/src/check/upvar.rs
@@ -2214,8 +2214,8 @@ fn determine_place_ancestry_relation<'tcx>(
 ///
 /// Reason we only drop the last deref is because of the following edge case:
 ///
-/// ```ignore HELP (I feel like this can be made to work but I'm doing something wrong)
-/// # struct A { field_of_a: i32 }
+/// ```
+/// # struct A { field_of_a: Box<i32> }
 /// # struct B {}
 /// # struct C<'a>(&'a i32);
 /// struct MyStruct<'a> {
@@ -2225,7 +2225,7 @@ fn determine_place_ancestry_relation<'tcx>(
 /// }
 ///
 /// fn foo<'a, 'b>(m: &'a MyStruct<'b>) -> impl FnMut() + 'static {
-///     let c = || drop(&*m.a.field_of_a);
+///     || drop(&*m.a.field_of_a)
 ///     // Here we really do want to capture `*m.a` because that outlives `'static`
 ///
 ///     // If we capture `m`, then the closure no longer outlives `'static'
```